src/file_reader.h

```diff
@@ -36,6 +36,12 @@ class BaseFileReader : public Reader {
     zsize_t size() const { return _size; };
     offset_t offset() const { return _offset; };
 
+#ifdef ENABLE_USE_MMAP
+    virtual const Buffer get_mmap_buffer(offset_t offset,
+                                         zsize_t size) const = 0;
+#endif
```

Comment:
This feels a little like making the ABI dependent on compilation options. Upon further contemplation it seems safe but I would prefer if the conditional compilation is restricted to the .cpp files.

Reply:
Done in fixup commit.